gen.go

```diff
@@ -891,6 +1001,161 @@ func renameMutation(s string) string {
+		}
+	case *types.TypeName:
+		exampleValue = "TYPENAME_TODO"
+	default:
+		exampleValue = "UNKNOWN_TODO"
+	}
+
+	return fmt.Sprintf(` example:"%s"`, exampleValue)
+}
+
+func yamlStructTag(fieldType *types.InputValueDefinition) string {
+	jsonStructTag := jsonStructTag(fieldType)
+	return strings.Replace(jsonStructTag, "json", "yaml", 1)
+}
+
+func jsonStructTag(fieldType *types.InputValueDefinition) string {
+	fieldName := fieldType.Name.Name
+	if isNullable(fieldType.Type) {
+		return fmt.Sprintf(`json:"%s,omitempty"`, fieldName)
+	}
+	return fmt.Sprintf(`json:"%s"`, fieldName)
+}
+
+func graphqlTypeToGolang(graphqlType string) string {
+	var convertedType string
+	nullable := "*Nullable["
+	if strings.HasSuffix(graphqlType, "!") {
+		graphqlType = strings.TrimSuffix(graphqlType, "!")
+	} else {
+		// GraphQL nullable --> Go pointer
+		convertedType += nullable
+	}
+
+	// GraphQL list --> Go slice
+	if strings.HasPrefix(graphqlType, "[") {
+		graphqlType = strings.TrimPrefix(graphqlType, "[")
+		graphqlType = strings.TrimSuffix(graphqlType, "]")
+		// NOTE: pretty sure we don't support slices containing pointers
+		graphqlType = strings.TrimSuffix(graphqlType, "!")
+		convertedType += "[]"
+	}
+
+	// GraphQL scalars/types --> Go type
+	switch graphqlType {
+	case "Boolean":
+		convertedType += "bool"
+	case "Float":
+		convertedType += "float64"
+	case "Int":
+		convertedType += "int"
+	case "ISO8601DateTime":
+		convertedType += "iso8601.Time"
+	case "String":
+		convertedType += "string"
+	default:
+		convertedType += graphqlType
+	}
+
+	if strings.HasPrefix(convertedType, nullable) {
+		convertedType += "]"
+	}
+
+	return convertedType
+}
+
+func getFieldTypeNew(fieldType types.Type) string {
+	return graphqlTypeToGolang(fieldType.String())
+}
+
+func getFieldTypeForObject(fieldType types.FieldDefinition) string {
+	goType := graphqlTypeToGolang(fieldType.Type.String())
+	if strings.HasPrefix(goType, "*Nullable[") {
+		goType = strings.TrimPrefix(goType, "*Nullable[")
+		goType = strings.TrimSuffix(goType, "]")
+	}
+
+	switch goType {
+	case "Filter":
+		return "FilterId"
+	case "JSONSchema":
+		return "JSON"
+	case "Service":
+		return "ServiceId"
+	case "Team":
+		goType = "TeamId"
+	case "User":
+		return "UserId"
+	}
+
+	if fieldType.Name == "AlertSource" && goType == "Integration" {
+		goType = "IntegrationId"
+	} else if fieldType.Name == "CustomActionsTriggerDefinition" && goType == "Action" {
+		goType = "CustomActionsId"
+	} else if fieldType.Name == "FilterPredicate" && goType == "CaseSensitive" {
+		goType = "*bool"
+	} else if fieldType.Name == "Property" {
+		switch goType {
+		case "[]Error":
+			goType = "[]OpsLevelErrors"
+		case "HasProperties":
+			goType = "EntityOwnerService"
+		case "JsonString":
+			goType = "*JsonString"
+		case "PropertyDefinition":
+			goType = "PropertyDefinitionId"
+		}
+	} else if fieldType.Name == "ServiceRepository" && goType == "Repository" {
+		goType = "RepositoryId"
+	}
+
+	return goType
+}
```

Comment:
This logic is pretty hard to follow.  Something to mark for refactoring or thinking if there is a better way.  I don't have a good handle on all the context needed to understand this yet but feels like a missing design pattern we could leverage that would significantly clean this up.  Maybe the Open/Closed principle or the Liskov Substitution principle.

Reply:
I'll work on this

Reply:
This function is not used in this PR. Resolving this thread.